sycl/doc/SYCL_compiler_and_runtime_design.md

```diff
@@ -0,0 +1,377 @@
+# SYCL\* Compiler and Runtime architecture design
+
+## Introduction
+
+This document describes the architecture of the SYCL compiler and runtime
+library.  Base SYCL specification version is
+[1.2.1](https://www.khronos.org/registry/SYCL/specs/sycl-1.2.1.pdf).
+
+## SYCL Compiler architecture
+
+SYCL application compilation flow:
+
+![High level component diagram for SYCL Compiler](Compiler-HLD.svg)
+
+SYCL compiler logically can be split into the host compiler and a number of
+device compilers - one per each supported target. Clang driver orchestrates the
```

Comment:
I have the feeling that here and in the following paragraphs most of your `-` here are actually emdashes instead `—`, except for the itemize syntax at the beginning of the lines.